source/extensions/filters/http/cache/http_cache_utils.cc

```diff
@@ -0,0 +1,175 @@
+#include "extensions/filters/http/cache/http_cache_utils.h"
+
+#include <array>
+#include <string>
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/strip.h"
+
+using absl::ascii_isalnum;
+using absl::ascii_isdigit;
+using absl::Duration;
+using absl::InfiniteDuration;
+using absl::InfinitePast;
+using absl::ParseTime;
+using absl::Seconds;
+using absl::SimpleAtoi;
+using absl::string_view;
+using absl::Time;
+using absl::ZeroDuration;
+using Envoy::Http::HeaderEntry;
+using std::array;
+using std::string;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace Internal {
+
+namespace {
+
+bool tchar(char c) {
+  switch (c) {
+  case '!':
+  case '#':
+  case '$':
+  case '%':
+  case '&':
+  case '*':
+  case '+':
+  case '-':
+  case '.':
+  case '^':
+  case '_':
+  case '`':
+  case '|':
+  case '~':
+    return true;
+  }
+  return ascii_isalnum(c);
+}
+
+bool eatToken(string_view& s) {
```

Comment:
describe what kind of token you are eating here, and also give an example of the type of syntax you are parsing.

Reply:
Done.